06-building-a-program-start-to-finish/solving-the-problem/output/app/class/invertedIndexClass.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * Inverted Index class
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * class constructor
+   * @constructor
+  */
+  constructor() {
+    this.indices = {};
+    this.indexedFiles = {};
+  }
+
+  readFile(fileToRead) {
+    const fileReader =  new FileReader();
+    // readFiles[filename] = [];
+    fileReader.readAsText(fileToRead);
+    return new Promise((resolve, reject) => {
+      fileReader.onload = () => {
+        try {
+          return resolve(JSON.parse(fileReader.result));
+        } catch (e) {
+          return reject('improperly formatted JSON file');
+        }
+      };
+    });
+  }
+
+  /**
+   * genereate indexes
+   * @function
+   * @param {Array} bookArray
+   * @param {string} title
+   * @returns {Object} indexes
+  */
+  createIndex(fileName, file) {
+    if (!file) { return false; }
+    const words = [];
+    const indexedWords = {};
+    file.forEach((book) => {
+      words.push(this.tokenize(book));
+    });
+    words.forEach((book, index) => {
+      for (let i = 0; i < book.length; i += 1) {
```

Comment:
This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)